src/context/TodoContext.tsx

```diff
@@ -4,46 +4,43 @@ import React, {
   createContext,
   useContext,
   useEffect,
-  useMemo,
-  useRef,
   useState,
 } from 'react';
 import { Todo } from '../types/Todo';
-import { ErrorList } from '../types/ErrorList';
+import { ErrorMessages } from '../types/ErrorMessages';
 import { Status } from '../types/Status';
 import { wait } from '../utils/fetchClient';
 import { deleteTodo, getTodos } from '../api/todos';
 
-export type TodoContext = {
+export type TodosContextType = {
   todos: Todo[];
-  preparedTodos: Todo[];
   setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
   tempTodo: Todo | null;
   setTempTodo: Dispatch<SetStateAction<Todo | null>>;
-  errorMessage: ErrorList | string;
-  setErrorMessage: Dispatch<SetStateAction<ErrorList | string>>;
+  errorMessage: ErrorMessages | string;
```

Comment:
You can create separate error message for empty string to remove `| string` code